New distributive rpm

siddharthist opened the pull request:
Fixes #633, #717, #740, enables #1045.
We will need to switch to CiscoCloud Bintray before merging.
Tested on Vagrant.

TanyaCouture commented:
Successfully installed on:
Vagrant
AWS

TanyaCouture commented:
Openstack installation:
The UI display says:
    Checks: Unknown
        No Health Checks loaded

ryane commented on roles/common/defaults/main.yml:
```diff
@@ -2,4 +2,5 @@
-selinux_state: permissive
\ No newline at end of file
+selinux_state: permissive
+distributive_rpm_url: "https://bintray.com/artifact/download/siddharthist/distributive-testing/distributive-0.2.4-1.x86_64.rpm"
```
will we be able to get this on an "official" bintray soon?

siddharthist replied:
I would love it if anyone on our team with access could just download it from there and reupload it to CiscoCloud.

ryane commented:
With this version, I am seeing failing checks on all control nodes for the following two services:

distributive-marathon-checks

```
time="2016-01-22T01:25:13Z" level=info msg="Creating checklist(s)..." path="/etc/distributive.d/marathon.json" type=file 
time="2016-01-22T01:25:13Z" level=fatal msg="Invalid check ID" id= 
```

distributive-mantlui-checks

```
time="2016-01-22T01:18:35Z" level=info msg="Creating checklist(s)..." path="/etc/distributive.d/mantlui.json" type=file 
time="2016-01-22T01:18:35Z" level=info msg="Running check File" 
time="2016-01-22T01:18:35Z" level=info msg="Running check DockerImageRegexp" 
time="2016-01-22T01:18:35Z" level=info msg="Running check PortTCP" 
time="2016-01-22T01:18:35Z" level=info msg="Running check PortTCP" 
time="2016-01-22T01:18:35Z" level=info msg="Running check File" 
time="2016-01-22T01:18:35Z" level=info msg="Running check File" 
time="2016-01-22T01:18:35Z" level=info msg="Running check File" 
time="2016-01-22T01:18:35Z" level=info msg="Report from checklist" checklist="mantlui health checks" report="↴\nTotal: 7\nPassed: 6\nFailed: 1\nOther: 0\nNo such file or directory: /etc/mantlui.nginx"
```

(/etc/mantlui.nginx no longer exists as of #955)

ryane commented:
also, once this is ready, should we just move it to the asteris bintray? A lot of our packages live there.

siddharthist commented:
Thanks for testing @ryane, I'll update the package and this PR.

siddharthist commented:
Rebased onto master, removed all marathon checks in preparation for adding them to the Marathon package. Works for me on Vagrant, needs further testing.

siddharthist commented:
Rebased and squashed, ready for testing! Works for me on Vagrant.

ryane commented:
I am getting this failure on all nodes (aws):

```
time="2016-02-01T21:34:22Z" level=info msg="Creating checklist(s)..." path="/etc/distributive.d/common.json" type=file 
time="2016-02-01T21:34:22Z" level=info msg="Running check Symlink" 
time="2016-02-01T21:34:22Z" level=info msg="Running check Installed" 
time="2016-02-01T21:34:22Z" level=info msg="Running check Installed" 
time="2016-02-01T21:34:22Z" level=info msg="Running check SystemctlUnitFileStatus" 
time="2016-02-01T21:34:22Z" level=info msg="Running check Installed" 
time="2016-02-01T21:34:22Z" level=info msg="Running check Installed" 
time="2016-02-01T21:34:22Z" level=info msg="Running check Directory" 
time="2016-02-01T21:34:22Z" level=info msg="Running check File" 
time="2016-02-01T21:34:22Z" level=info msg="Running check CPUUsage" 
time="2016-02-01T21:34:22Z" level=info msg="Running check MemoryUsage" 
time="2016-02-01T21:34:22Z" level=info msg="Running check DiskUsage" 
time="2016-02-01T21:34:25Z" level=info msg="Report from checklist" checklist="common health check" report="↴\nTotal: 11\nPassed: 10\nFailed: 1\nOther: 0\nUnit file could not be found: firewalld.service"
```

siddharthist commented:
I think I see the problem. I'll try again. Thanks once again @ryane!

TanyaCouture commented:
Successful installation on:
Vagrant

siddharthist commented:
Tested, working on Openstack

ryane commented:
+1 on aws
